### PR TITLE
Captcha protected faucet

### PR DESCRIPTION
**Pull request by rzadp**
Just a draft showcasing how we can use a new (unreleased) [external faucet](https://github.com/paritytech/substrate-matrix-faucet/pull/192) endpoint inside the docs.

Of course would need some more styling to make it nicer.

Screenshot 2023-02-02 at 09 17 18

### CODE REVIEW

**Comment by cmichi** [comment]
Awesome, thanks for picking up the work on this, @rzadp !

**Comment by rzadp** [comment]
If you're getting this when trying it out locally:

Screenshot 2023-02-09 at 10 16 05

You can do the following:

```bash
sudo bash -c 'echo "127.0.0.1 dev.use.ink" >> /etc/hosts'
```

And then open `http://dev.use.ink:8080/` instead of `http://localhost:8080/` in the browser and then it should work.

File: docs/testnet/Faucet.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, {useState} from "react";
+import ReCAPTCHA from "react-google-recaptcha";
+
+const RECAPTCHA_SITE_KEY = "6LdU5kckAAAAANktvvAKJ0auYUBRP0su94G7WXwe"
```
Comment (cmichi) [inline docs/testnet/Faucet.tsx:4]: Is it fine to have this key public?

Comment (rzadp) [reply docs/testnet/Faucet.tsx:4]: Yes, this is not a secret.

**Comment by cmichi** [comment]
It works perfectly, thank you so much!

**Review by cmichi — APPROVED** [review]

**Pull request merged**